react/lesson1.md

```diff
@@ -36,175 +91,431 @@ cd my-app/
+
+Child or `presentational` components are essentially smaller customisable to the data that they receive but usually do not have the data in themselves. These are often referred to as `dumb` components - because they do not process the data in themselves:
+
+While function components are a lot simpler to read and write, using ES6 classes allows us to create more powerful, smarter components.
+
+# JSX
+
+You might have noticed something weird: we were returning some HTML-like code in our components.
+
+That is called JSX, and it's a compromise to allows us to use the HTML we are so familiar inside our JavaScript components. 
+
+Since it's not exactly HTML, there are some gotchas, of which you'll need to be aware of:
+
+- event handlers are capitalized
+
+HTML: `<button onclick="myFunction()"/>` 
+JSX: `button onClick="myFunction()"/>`
+
+- `className` instead of `class`
 
-export default App;
+HTML: `<button class="myButton"/>`
+JSX: `<button className="myButton"/>`
 
+- must always have one child element
+
+Won't work:
+
+```javascript
+return (
+  <p> Hello! </p>
+  <p> This is my blog. </p>
+);
 ```
 
-Child or `presentational` components are essentially smaller customisable to the data that they receive but usually do not have the data in themselves. These are often referred to as `dumb` components - because they do not process the data in themselves:
+Will work:
 
-```jsx
-import React, { Component } from 'react';
+```javascript
+return (
+  <div>
+    <p> Hello! </p>
+    <p> This is my blog. </p>
+  </div>
+);
+```
 
-class Organisation extends Component {
-  render() {
-    const name = this.props.name;
-    const website = this.props.website;
-    return (
-        <div>
-            Name: {name} <br />
-            Website: {website}
-        </div>
-    );
+## Thinking in React
+
+Let's go over some examples and discuss how we would break everything up into smart and dumb components.
+
+# Event handlers
+
+Time to handle a few user events. We'll add a "difficulty setting" input. For now, whenever the user changes that input, we will `console.log` the result.
+
+```javascript
+
+difficultyChange = (event) => {
+  console.log(event.target.value);
+};
+
+//inside render()
+<input type="number" onChange={this.difficultyChange}/>
+```
+
+3 things to notice:
+
+- the curly brackets after `onChange`: surrounding something by two curly brackets is a way to use JavaScript inside JSX
+- `difficultyChange` is written as an arrow function: this is a good way to avoid issues with the `this` context in React. You can read more about it here: https://facebook.github.io/react/docs/handling-events.html
+- the `event` object: this is a special object passed to all event handlers in React. In our case we can use it to access the target DOM element of that event, and more specifically, its value.
+
+# JSX: Map Arrays to Components
+
+Because works so well with JavaScript, we can even it in for loops and build JS arrays that contain JSX elements.
```

Comment:
Not sure exactly how this sentence is intended to read

Reply:
Haha, me neither. I think it should've read: "Because JSX works so well with JavaScript, we can even use it in `for` loops and build JavaScript arrays that contain JSX elements."